src/plfa/part2/DeBruijn.lagda.md

```diff
@@ -414,33 +425,42 @@ The final term represents the Church numeral two.
 
 We can use a natural number to select a type from a context:
 ```
-lookup : Context → ℕ → Type
-lookup (Γ , A) zero     =  A
-lookup (Γ , _) (suc n)  =  lookup Γ n
-lookup ∅       _        =  ⊥-elim impossible
-  where postulate impossible : ⊥
+lookup : (Γ : Context) → (n : ℕ) → (p : n < length Γ) → Type
+lookup (_ , A) zero    (s≤s z≤n)  =  A
+lookup (Γ , _) (suc n) (s≤s p)    =  lookup Γ n p
```

Comment:
I may be worth emphasizing that a proof `n < length Γ` contains enough information to be used in place of the index `n`, and so you could let `n` be an *implicit* argument, and instead define `lookup` and `count` by induction on the proof `p`.

Reply:
That's sounds like a reasonable suggestion.

I've updated the corresponding functions to have `n` as an implicit argument.